pkg/server/job.go

```diff
@@ -0,0 +1,41 @@
+package server
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"os/exec"
+
+	pb "github.com/rexposadas/teleport/api"
+)
+
+// Job represent an individual job.
+type Job struct {
```

Comment:
I feel like the Job status potentially suffers from the risk of concurrent read/writes. Does this need a mutex over it?